daemon/networkdriver/ipallocator/allocator.go

```diff
@@ -41,19 +41,24 @@ var (
 	ErrBadSubnet                = errors.New("network does not contain specified subnet")
 )
 
-var (
-	lock         = sync.Mutex{}
-	allocatedIPs = networkSet{}
-)
+type IPAllocator struct {
+	allocatedIPs networkSet
+	sync.Mutex
```

Comment:
definitely, I actually meant to not expose it but I forgot about it at the end :pensive: